### PR TITLE
Journalist can add image to article

### PR DESCRIPTION
**Pull request by kermit-klein**
[PT LINK](https://www.pivotaltracker.com/story/show/172888059)

- Add image test to create article feature test (happy and sad path)
- Add sad path fixture for image upload
- Add an image for testing
- Add file_upload command to cypress
- Add preview uploaded file function
Screenshot from 2020-05-26 19-45-51

### CODE REVIEW

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 1]
```diff
@@ -58,10 +60,25 @@ describe("Journalist can create an article", () => {
...
     cy.get("input#title").type("This is the title");
     cy.get("#post").click();
     cy.get("#message").should("contain", "Body can't be blank");
   });
+
+  it("unsuccessfully without uploading image", () => {
+    cy.route({
+      method: "POST",
+      url: "http://localhost:3000/api/articles*",
+      response: "fixture:image_message.json",
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: please rename this fixture file, not so descriptive of when it is used

File: src/components/CreateArticle.jsx [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+
+  const toBase64 = (file) =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = () => resolve(reader.result);
+      reader.onerror = (error) => reject(error);
+    });
+
+  const handleChange = (e) => {
```
Comment (oliverochman) [inline src/components/CreateArticle.jsx]: rename this method
what are we changing?

File: src/components/CreateArticle.jsx [hunk 3]
```diff
@@ -21,9 +43,28 @@ const CreateArticle = () => {
...
+          onSubmitHandler={onSubmitHandler}
+          handleChange={handleChange}
+          message={message}
+        />
+      </Container>
+      <Container className="writing-container">
+        {imagePreview === "" ? null : (
+          <>
+            <p>Image Preview</p>
+            <Image
```
Comment (oliverochman) [inline src/components/CreateArticle.jsx:58]: cool👍🏼

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by PaulineBA — APPROVED** [review]

**Review by oliverochman — APPROVED** [review]
Good job!

File: cypress/support/commands.js [hunk 4]
```diff
@@ -1,25 +1,21 @@
...
+          const el = subject[0];
+          const testFile = new win.File([blob], name, { type });
+          const dataTransfer = new win.DataTransfer();
+          dataTransfer.items.add(testFile);
+          el.files = dataTransfer.files;
+          cy.wrap(subject).trigger("change", { force: true });
+        })
+    )
+  )
+})
```
Comment (jysmys) [inline cypress/support/commands.js:21]: najs shit!!
I like 💯

**Pull request merged**